compiler/rustc_resolve/src/lib.rs

```diff
@@ -1521,6 +1533,101 @@ impl<'a> Resolver<'a> {
         });
     }
 
+    /// Compute access levels for exports and intermediate use statements
+    fn resolve_access_level(&mut self, krate: &Crate) {
+        self.set_access_level(CRATE_NODE_ID, Some(AccessLevel::Public));
+
+        self.set_exports_access_level(CRATE_DEF_ID);
+        let mut privacy_visitor =
+            access_levels::PrivacyVisitor::new(self, Some(AccessLevel::Public));
+
+        // We need a double pass to correctly resolve exports
+        // as an item can be defined in a private module
+        // and exported later on
+        //
+        // mod foo {
+        //     pub enum Hungry;
+        // }
+        //
+        // pub mod bar {
+        //     pub use super::foo::*;
+        // }
+
+        visit::walk_crate(&mut privacy_visitor, krate);
+        visit::walk_crate(&mut privacy_visitor, krate);
+
+        tracing::info!("resolve::access_levels: {:#?}", self.access_levels);
+    }
+
+    fn set_exports_access_level(&mut self, module_id: LocalDefId) {
+        if let Some(exports) = self.export_map.get(&module_id) {
+            let pub_exports = exports
+                .iter()
+                .filter(|ex| ex.vis == Visibility::Public)
```

Comment:
`pub` exports are only `Public`/`Exported` if the `module_id` module is `Public`/`Exported` respectively, I don't see this condition here.